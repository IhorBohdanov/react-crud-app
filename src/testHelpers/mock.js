export const albumData = {
  id: 1,
  title: "Album title",
  userId: 1,
};

export const photos = [
  {
    albumId: 1,
    id: 1,
    title: "accusamus beatae ad facilis cum similique qui sunt",
    url: "https://via.placeholder.com/600/92c952",
    thumbnailUrl: "https://via.placeholder.com/150/92c952",
  },
  {
    albumId: 1,
    id: 2,
    title: "reprehenderit est deserunt velit ipsam",
    url: "https://via.placeholder.com/600/771796",
    thumbnailUrl: "https://via.placeholder.com/150/771796",
  },
  {
    albumId: 1,
    id: 3,
    title: "officia porro iure quia iusto qui ipsa ut modi",
    url: "https://via.placeholder.com/600/24f355",
    thumbnailUrl: "https://via.placeholder.com/150/24f355",
  },
];

export const albums = [
  {
    id: 1,
    title: "quidem molestiae enim",
    userId: 1,
  },
  {
    id: 2,
    title: "sunt qui excepturi placeat culpa",
    userId: 2,
  },
];

export const newAlbum = {
  id: 3,
  title: "Album title",
  userId: 2,
}

export const editedAlbum = {
  id: 1,
  title: "Edited album title",
  userId: 2,
}