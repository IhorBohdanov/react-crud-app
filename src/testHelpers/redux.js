import React from "react";
import { Provider } from "react-redux";
import { compose, createStore, applyMiddleware } from "redux";
import { rootReducer } from "../store/rootReducer";
import { sagaWatcher } from "../store/sagas";
import createSagaMiddleware from "redux-saga";

const saga = createSagaMiddleware();
const store = createStore(rootReducer, compose(applyMiddleware(saga)));

export const wrapWithRedux = (component) => {
  saga.run(sagaWatcher);
  return <Provider store={store}>{component}</Provider>;
};
