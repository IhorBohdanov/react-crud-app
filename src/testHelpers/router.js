import { Router, Route, Switch } from "react-router-dom";
import { createMemoryHistory } from "history";

export const wrapWithRout = (component, { path, route }) => {
  const history = createMemoryHistory();
  history.push(route);

  return (
    <Router history={history}>
      <Switch>
        <Route path={path}>
          {component}
        </Route>
      </Switch>
    </Router>
  );
};
