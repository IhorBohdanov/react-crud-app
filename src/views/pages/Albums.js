import React from 'react';
import { List } from 'antd';
import AlbumItem from '../components/AlbumItem';

export default function Albums({albums, onAlbumDelete}) {
  return (
    <div>
      <h1>Albums page</h1>
      {
        albums.length ?
          (
            <List
              dataSource={albums}
              renderItem={item => (
                <AlbumItem item={item} key={item.id} handleDelete={onAlbumDelete}/>
              )}
            />
          )
          : "No data"
      }
    </div>
  )
}
