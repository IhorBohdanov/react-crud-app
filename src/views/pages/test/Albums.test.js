import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from 'react-router-dom';
import Albums from "../Albums";
import { albums } from "../../../testHelpers";



describe("Albums page", () => {
  let onAlbumDelete;

  beforeEach(() => {
    onAlbumDelete =jest.fn();
  })

  it('match snapshot', () => {
    const { asFragment } = render(<Albums albums={[]} onAlbumDelete={onAlbumDelete} />);
    expect(asFragment(<Albums albums={[]} onAlbumDelete={onAlbumDelete} />)).toMatchSnapshot();
  });

  it("Render empty Albums page", () => {
    render(<Albums albums={[]} onAlbumDelete={onAlbumDelete} />);
    const heading = screen.getByRole("heading", { name: "Albums page" });
    const noDataLabel = screen.getByText(/no data/i);
    expect(heading).toBeInTheDocument();
    expect(noDataLabel).toBeInTheDocument();
  });

  it("Render list of albums", () => {
    render(<Router><Albums albums={albums} onAlbumDelete={onAlbumDelete} /></Router>);
    const albumsList = screen.getAllByRole('listitem');
    expect(albumsList).toHaveLength(2);
  });
});
