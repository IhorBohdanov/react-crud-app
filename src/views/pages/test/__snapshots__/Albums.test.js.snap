// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums page match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Albums page
    </h1>
    No data
  </div>
</DocumentFragment>
`;

exports[`Albums page should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Albums page
    </h1>
    No data
  </div>
</DocumentFragment>
`;
